metadatahandler.py

```diff
@@ -134,56 +135,53 @@ def writeGamelistEntry(self, gamelist, dosGame, game, genre, useGenreSubFolders,
     def buildGenre(self, dosGame):
-        if dosGame is not None and dosGame.genres is not None:
-            if 'Flight Simulator' in dosGame.genres or 'Vehicle Simulation' in dosGame.genres:
-                return 'Simulation'
-            elif "Education" in dosGame.genres or "Quiz" in dosGame.genres:
-                if "Adventure" in dosGame.genres or "Visual Novel" in dosGame.genres:
-                    return "Adventure-Visual"
-                else:
-                    return 'Misc'
-            elif "Racing" in dosGame.genres or "Driving" in dosGame.genres or "Racing / Driving" in dosGame.genres:
-                return "Race"
-            elif 'Sports' in dosGame.genres:
-                return 'Sports'
-            elif 'Pinball' in dosGame.genres:
-                return 'Pinball'
-            elif "Puzzle" in dosGame.genres or "Board" in dosGame.genres or "Board / Party Game" in dosGame.genres \
-                    or "Casino" in dosGame.genres or 'Cards / Tiles' in dosGame.genres or 'Game Show' in dosGame.genres:
-                return "Puzzle"
-            elif 'Shooter' in dosGame.genres:
-                return 'ShootEmUp'
-            elif 'Platform' in dosGame.genres:
-                return 'Platform'
-            elif 'FPS' in dosGame.genres or 'First Person Shooter' in dosGame.genres:
-                return 'Gun-FPS'
-            elif 'Fighting' in dosGame.genres or 'Beat \'em Up' in dosGame.genres:
-                return 'BeatEmUp'
-            elif 'Strategy' in dosGame.genres and "Puzzle" not in dosGame.genres:
-                return 'Strategy-Gestion'
-            elif 'RPG' in dosGame.genres or 'Role-Playing' in dosGame.genres:
-                return 'RPG'
-            elif 'Interactive Fiction' in dosGame.genres:
-                return "Adventure-Visual"
-            elif "Adventure" in dosGame.genres and "Action" in dosGame.genres:
-                return "Action-Adventure"
-            elif "Adventure" in dosGame.genres or "Visual Novel" in dosGame.genres:
-                return "Adventure-Visual"
-            elif 'Simulation' in dosGame.genres and 'Managerial' in dosGame.genres:
-                return 'Strategy-Gestion'
-            elif 'Construction and Management Simulation' in dosGame.genres:
-                return 'Strategy-Gestion'
-            elif 'Simulation' in dosGame.genres:
-                return 'Simulation'
-            elif 'Shooter' in dosGame.genres:
-                return 'ShootEmUp'
-            elif 'Action' in dosGame.genres:
-                return 'Action-Adventure'
-            elif 'Arcade' in dosGame.genres or 'Life Simulation' in dosGame.genres:
-                return 'Misc'
-            elif 'Creativity' in dosGame.genres or 'App' in dosGame.genres or 'Reference' in dosGame.genres:
-                return 'Tools'
-            else:
-                return 'Unknown'
-        else:
-            return 'Unknown'
+        
```

Comment:
I meant also moving this function in genre_mapping.py and calling it here ;)

Reply:
Gotcha - check now :)

We're down to 5 diffs with eXo 5.0 :

<pre>
+---+----------+-------------------------------------------------------------------+------------------+---------------------+
| 0 | universa | ['Action', 'Construction and Management Simulation', 'Shooter']   | ShootEmUp        | Strategy-Management |
| 1 | eradicat | ['Action', 'First Person Shooter', 'Puzzle']                      | Puzzle           | Gun-FPS             |
| 2 | corporat | ['Action', 'First Person Shooter', 'Puzzle']                      | Puzzle           | Gun-FPS             |
| 3 | monteret | ['Action', 'First Person Shooter', 'Platform', 'Puzzle']          | Puzzle           | Gun-FPS             |
| 4 | execsui  | ['Construction and Management Simulation', 'Interactive Fiction'] | Adventure-Visual | Strategy-Management |
+---+----------+-------------------------------------------------------------------+------------------+---------------------+
</pre>